presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PlanNodeSearcher.java

```diff
@@ -29,18 +30,32 @@
 
 public class PlanNodeSearcher
 {
+    @Deprecated
```

Comment:
I am not sure if it is the way to go to use `PlanNodeSearcher` in the iterative optimizer. I think this rule depends on having pattern matching and traits to be implemented first.

@martint What do you think?